numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -709,47 +707,32 @@ VOID_getitem(void *input, void *vap)
                 "tried to get void-array with object members as buffer.");
         return NULL;
     }
-    itemsize = PyArray_DESCR(ap)->elsize;
 
-#if defined(NPY_PY3K)
-    /*
-     * Return a byte array; there are no plain buffer objects on Py3
-     */
+    /* Return a memoryview */
     {
-        npy_intp dims[1], strides[1];
-        dims[0] = itemsize;
-        strides[0] = 1;
-        descr = PyArray_DescrNewFromType(NPY_BYTE);
-        u = (PyArrayObject *)PyArray_NewFromDescr(&PyArray_Type,
-                             descr, 1, dims, strides, ip,
+        PyObject *ret, *u;
+
+        /* first create a size-1 view of this element */
+        npy_intp dims[1];
+        dims[0] = 1;
+        Py_INCREF(descr);
+        u = PyArray_NewFromDescr(&PyArray_Type, descr, 1, dims, NULL, ip,
```

Comment:
Could make this PEP8 compliant. Starting a new line after the opening `(` might be the easiest way.